includes/sanitizers/class-amp-nav-menu-toggle-sanitizer.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Class AMP_Nav_Menu_Toggle_Sanitizer
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Nav_Menu_Toggle_Sanitizer
+ *
+ * Handles state for navigation menu toggles, based on theme support.
+ *
+ * @since 1.1.0
+ */
+class AMP_Nav_Menu_Toggle_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Default args.
+	 *
+	 * @since 1.1.0
+	 * @var array
+	 */
+	protected $DEFAULT_ARGS = array(
+		'nav_container_id'           => '',
+		'nav_container_xpath'        => '', // Alternative for 'nav_container_id', if no ID available.
+		'menu_button_id'             => '',
+		'menu_button_xpath'          => '', // Alternative for 'menu_button_id', if no ID available.
+		'nav_container_toggle_class' => '',
+		'menu_button_toggle_class'   => '', // Optional.
+		'nav_menu_toggle_state_id'   => 'navMenuToggledOn',
+	);
+
+	/**
+	 * XPath.
+	 *
+	 * @since 1.1.0
+	 * @var DOMXPath
+	 */
+	protected $xpath;
+
+	/**
+	 * AMP_Nav_Menu_Toggle_Sanitizer constructor.
+	 *
+	 * @since 1.1.0
+	 *
+	 * @param DOMDocument $dom  DOM.
+	 * @param array       $args Args.
+	 */
+	public function __construct( $dom, $args = array() ) {
+		parent::__construct( $dom, $args );
+
+		// Ensure the state ID is always set.
+		if ( empty( $this->args['nav_menu_toggle_state_id'] ) ) {
+			$this->args['nav_menu_toggle_state_id'] = $this->DEFAULT_ARGS['nav_menu_toggle_state_id'];
+		}
+	}
+
+	/**
+	 * If supported per the constructor arguments, inject `amp-state` and bind dynamic classes accordingly.
+	 *
+	 * @since 1.1.0
+	 */
+	public function sanitize() {
+		$theme_support_args = AMP_Theme_Support::get_theme_support_args();
```

Comment:
This needs to be looking at `$this->args` not theme support. Obtaining the global state here will be problematic for response caching.